compiler/record-minmax/CMakeLists.txt

```diff
@@ -19,8 +22,16 @@ add_executable(record-minmax ${DRIVER} ${SOURCES})
 target_include_directories(record-minmax PRIVATE include)
 target_include_directories(record-minmax PRIVATE ${Boost_INCLUDE_DIRS})
 target_include_directories(record-minmax PRIVATE ${HDF5_INCLUDE_DIRS})
-
-target_link_libraries(record-minmax ${HDF5_CXX_LIBRARIES})
+unset(HDF5_STATIC_LIBRARIES)
+foreach(LIB ${HDF5_CXX_LIBRARIES})
+  if(LIB MATCHES libsz.so OR LIB MATCHES libz.so)
+    string(REGEX REPLACE "[.]so$" ".a" LIB ${LIB})
```

Comment:
As this does not apply to Windows platform, how about extracting this block of new codes to a function or a macro and call it. inside the macro or function it can branch depending on the platform
(I'm not sure branching by the platform is good or bad)

Reply:
@mhs4670go, @seanshpark 

1) I guessed it would work on windows because we're using mingw or something for windows build, not native windows at least last Dec. Maybe I miss something. Could you let me know why this does nt apply to window platform?

@mhs4670go

2) As I know, `find_package` should have searched `*.a` if `HDF5_USE_STATIC_LIBRARIES` is set before `find_package`.

Reply:
This is cache problem. Even if I set `HDF5_USE_STATIC_LIBRARIES`, cmake doesn't set it `*.a` because it's already `*.so`.